packages/grid/_modules_/grid/components/toolbar/GridToolbarFilterButton.tsx

```diff
@@ -30,85 +30,96 @@ const useStyles = makeStyles(
-                <li key={item.id}>
-                  {`${lookup[item.columnField!].headerName || item.columnField}
+    return (
+      <div>
+        {apiRef!.current.getLocaleText('toolbarFiltersTooltipActive')(counter)}
+        <ul className={classes.list}>
+          {activeFilters.map((item) => ({
+            ...(lookup[item.columnField!] && (
+              <li key={item.id}>
+                {`${lookup[item.columnField!].headerName || item.columnField}
                   ${getOperatorLabel(item)}
                   ${item.value}`}
-                </li>
-              )),
-            }))}
-          </ul>
-        </div>
-      );
-    }, [apiRef, preferencePanel.open, counter, activeFilters, lookup, classes]);
-
-    const toggleFilter = React.useCallback(() => {
-      const { open, openedPanelValue } = preferencePanel;
-      if (open && openedPanelValue === GridPreferencePanelsValue.filters) {
-        apiRef!.current.hideFilterPanel();
-      } else {
-        apiRef!.current.showFilterPanel();
-      }
-    }, [apiRef, preferencePanel]);
+              </li>
+            )),
+          }))}
+        </ul>
+      </div>
+    );
+  }, [apiRef, preferencePanel.open, counter, activeFilters, lookup, classes]);
 
-    // Disable the button if the corresponding is disabled
-    if (options.disableColumnFilter) {
-      return null;
+  const toggleFilter = React.useCallback(() => {
+    const { open, openedPanelValue } = preferencePanel;
+    if (open && openedPanelValue === GridPreferencePanelsValue.filters) {
+      apiRef!.current.hideFilterPanel();
+    } else {
+      apiRef!.current.showFilterPanel();
     }
+  }, [apiRef, preferencePanel]);
 
-    const OpenFilterButtonIcon = apiRef!.current.components!.OpenFilterButtonIcon!;
-    return (
-      <Tooltip title={tooltipContentNode} enterDelay={1000}>
-        <Button
-          ref={ref}
-          {...props}
-          onClick={toggleFilter}
-          size="small"
-          color="primary"
-          aria-label={apiRef!.current.getLocaleText('toolbarFiltersLabel')}
-          startIcon={
-            <Badge badgeContent={counter} color="primary">
-              <OpenFilterButtonIcon />
-            </Badge>
-          }
-        >
-          {apiRef!.current.getLocaleText('toolbarFilters')}
-        </Button>
-      </Tooltip>
-    );
-  },
-);
+  // Disable the button if the corresponding is disabled
+  if (options.disableColumnFilter) {
+    return null;
+  }
+
+  const OpenFilterButtonIcon = apiRef!.current.components!.OpenFilterButtonIcon!;
+  return (
+    <Tooltip title={tooltipContentNode} enterDelay={1000} {...other}>
+      <Button
+        ref={ref}
+        size="small"
+        color="primary"
+        aria-label={apiRef!.current.getLocaleText('toolbarFiltersLabel')}
+        startIcon={
+          <Badge badgeContent={counter} color="primary">
+            <OpenFilterButtonIcon />
+          </Badge>
+        }
+        onClick={toggleFilter}
```

Comment:
Either call `componentProps.button?.onClick?.()` in `toggleFilter` or omit `toggleFilter` from the `componentProps.button` interface.

@oliviertassinari do we want to allow the override of any prop ?
For instance here, the user could override `startIcon` which is already customizable with the `OpenFilterButtonIcon` (but he currenctly can't remove the badge).

Reply:
Not an "easy" change after all ;)
IMO we should be able to change completely the appearance of the button as long as the behavior is kept.

Reply:
I'm also in favor of letting the user change the  `startIcon`, and it's a modification the user could do by mistake.

Reply:
> do we want to allow the override of any prop ?

@flaviendelangle Yes, I think so. Developers should be able to override `startIcon`.

The case of the event handlers is interesting. We don't have a policy around it yet. In the core, If I look at `SliderUnstyled.js`, `componentProps` overrides the event handlers. If I look at `SwitchUnstyled`, the event handlers of `componentProps` are merged.

cc @michaldudak and @mnajdova. Maybe we should merge?

Reply:
We should definitively have consistency in this matter across the codebase. My preference for event handlers and classNames is to merge. I suppose it'll create less WTF moments among developers 😉

Reply:
@flaviendelangle I have noticed that you merged without merging the onClick handlers. Was it on purpose? Is this something we should handle in a follow-up?

Reply:
I didn't see you already merged, I may add another PR to deal with event merging.

Reply:
> I may add another PR to deal with event merging.

@tifosiblack Flavien is off until Monday, another PR he can review by the time he comes back would be awesome 👌

Reply:
Definitely an error on my side